### DISCUSSION TITLE
Developer to hacker

### DISCUSSION THREAD

**Discussion by Orionzbelt** [LyleMi/Learn-Web-Hacking#16] [General]
I've always found hacking fascinating. Right from when I was a kid. Now that I'm grown and have made up my mind to switch carriers by next year, I've started learning how to code. Starting with the front end, then back end and finally to data analytics and structures. I Was told if I go through this process I will be an all round hacker when I decide to go into that field finally. 

My question are these;
Do I necessary need to learn Web development before hacking?? 

If yes, do I need to start from front end or back end??

**Comment by LyleMi** [comment]
The first thing I want to say is that hacking is wide-ranging. From the perspective of CTF, it can be divided into pwn web crypto misc and many others. 

From the view of web security alone, learning web development is not necessarily necessary but meaningful. It can provide you with a new perspective when doing security analysis, the developer's perspective. This perspective can make it easier for you to know which part of the code is more vulnerable.

In addition, penetration testing and code auditing are both hacking, but development in a broad sense is also hacking. You don't have to crack things, you can build things.

Finally, it is okay to learn front-end or back-end first, follow your hobbies. Wishing you a pleasant experience in your hacking journey.
